fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -244,6 +244,20 @@ class RestoreConfig : public KeyBackedConfig {
 
 	Key applyMutationsMapPrefix() { return uidPrefixKey(applyMutationsKeyVersionMapRange.begin, uid); }
 
+	ACTOR static Future<Version> getCurrentVersion_impl(Reference<ReadYourWritesTransaction> tr, UID uid) {
```

Comment:
`getCurrentVersion` is a misleading name for this function.   If a backup is completed, it still has a version to which it was completed, but this function will not return that and so restore status will not report that.

This function is simply returning the apply begin version - this is the version an active mutations apply stream has been completed to by the cluster.   There is a function which sets this, called `setApplyBeginVersion()`, so this new function should be called `getApplyBeginVersion()` and be defined next to that, a bit further down in this file.  It should return an Optional<Version>, which will be present if the key exists in the database.

Then, `getCurrentVersion()` can use `getApplyBeginVersion()` when appropriate.  
- If the restore is in the `running` state, then the current version is the `getApplyBeginVersion()`
- If the restore is in the `completed` state, the current version is the restore target version which comes from the `restoreVersion()` property.
- If the restore is in any other state, the current version can be reported as -1 as you have done.

Reply:
Done. I have a question: in getCurrentVersion_impl, if I write:
```
state Version version = -1;
if (status == ERestoreState::RUNNING) {
	version = wait(self->getApplyBeginVersion(tr));
} else if (status == ERestoreState::COMPLETED) {
	version = wait(self->restoreVersion().getD(tr));
}
return version;
```
I will get `Declaration has no type` error. Do you know why?

Reply:
Yes, it's a quirk of Flow that the result of a `wait()` has to be assigned to new local or state variable.  One way around this is to use the generic actor `store()` which would look like 
`wait(store(version, self->getApplyBeginVersion(tr)));`
...
`wait(store(version, self->restoreVersion().getD(tr)));`